packages/utils/upgrade/src/modules/project/__tests__/project.test.ts

```diff
@@ -108,7 +115,9 @@ describe('Project', () => {
     test('Succeed for valid AppProject', () => {
       vol.fromNestedJSON(appVolume, defaultCWD);
 
-      const project = projectFactory(defaultCWD) as AppProject;
+      const project: IAppProject | IPluginProject = projectFactory(defaultCWD);
```

Comment:
That would be `: IProject`, no?

Since it's a discriminated union of interface, it should accept both (also, I'm not even sure we need to type it, it should be able to infer it automatically)

Reply:
oops, that got left in while I was debugging the assertions, removed!